common/transcript.go

```diff
@@ -20,21 +22,22 @@ func NewTranscript(label string) *Transcript {
 
 	transcript := &Transcript{
 		state: digest,
+		buff:  bytes.NewBuffer(make([]byte, 0, 1024)),
 	}
 
 	return transcript
 }
 
-func (t *Transcript) AppendMessage(message []byte, label string) {
-	t.state.Write([]byte(label))
-	t.state.Write(message)
+func (t *Transcript) AppendMessage(message []byte, label []byte) {
```

Comment:
I changed the parameter type from `string` to `[]byte`, since this avoid having to do allocations  in the removed L29 (i.e: `t.state.Write([]byte(label))` has to do an allocation to transform a `string` (immutable) to `[]byte` (mutable); which is unfortunate).